fuel-vm/src/tests/memory.rs

```diff
@@ -181,3 +218,157 @@ fn dynamic_call_frame_ops() {
         initial_sp + STACK_EXTEND_AMOUNT as u64 - STACK_SHRINK_AMOUNT as u64
     );
 }
+
+#[rstest::rstest]
+fn test_mcl_and_mcli(
+    #[values(0, 1, 7, 8, 9, 255, 256, 257)] count: u32,
+    #[values(true, false)] half: bool, // Clear only first count/2 bytes
+    #[values(true, false)] mcli: bool, // Test mcli instead of mcl
+) {
+    // Allocate count + 1 bytes of memory, so we can check that the last byte is not cleared
+    let mut ops = vec![op::movi(0x10, count + 1), op::aloc(0x10), op::movi(0x11, 1)];
+    // Fill with ones
+    for i in 0..(count + 1) {
+        ops.push(op::sb(RegId::HP, 0x11, i as u16));
+    }
+    // Clear it, or only half if specified
+    if mcli {
+        if half {
+            ops.push(op::mcli(RegId::HP, count / 2));
+        } else {
+            ops.push(op::mcli(RegId::HP, count));
+        }
+    } else {
+        ops.push(op::movi(0x10, count));
+        if half {
+            ops.push(op::divi(0x10, 0x10, 2));
+        }
+        ops.push(op::mcl(RegId::HP, 0x10));
+    }
+    // Log the result and return
+    ops.push(op::movi(0x10, count + 1));
+    ops.push(op::logd(0, 0, RegId::HP, 0x10));
+    ops.push(op::ret(RegId::ONE));
+
+    let vm = setup(ops);
+    let vm: &Interpreter<MemoryStorage, Script> = vm.as_ref();
+
+    if let Some(Receipt::LogData { data, .. }) = vm.receipts().first() {
+        let c = count as usize;
+        assert_eq!(data.len(), c + 1);
```

Comment:
Might be helpful to have some comments on these assertions to explain what's being tested.